parlai/core/dict.py

```diff
@@ -493,6 +497,115 @@ def act(self):
                         self.add_to_dict(self.tokenize(text))
         return {'id': 'Dictionary'}
 
+    def vectorize(self, obs, mode='train', use_cuda=True):
+        """
+        Converts 'text' and 'label'/'eval_label' field to vectors
+        kwargs:
+        -mode: which label should be passed through, is either train, eval or test
+        -use_cuda: whether to use cuda
+        """
+        if 'text' not in obs:
+            return obs
+        # convert 'text' field to vector using self.txt2vec and then a tensor
+        obs['text'] = torch.LongTensor(self.txt2vec(obs['text']))
+        if use_cuda:
+            obs['text'] = obs['text'].cuda()
+
+        if mode == 'train':
+            assert 'labels' in obs
```

Comment:
this assertion only applies to some models, e.g. unsupervised training may not use text-label pairs but could still want to train on the text.

I think actually you can cut `mode` from the arguments completely and just parse the labels or the eval labels if they are there, otherwise not.